public/app/core/components/Login/LoginCtrl.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import appEvents from 'app/core/app_events';
+import config from 'app/core/config';
+
+import { updateLocation } from 'app/core/actions';
+import { connect } from 'react-redux';
+import { StoreState } from 'app/types';
+import { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import { hot } from 'react-hot-loader';
+
+export interface FormModel {
+  user: string;
+  password: string;
+  email: string;
+}
+interface Props {
+  routeParams?: any;
+  updateLocation?: typeof updateLocation;
+  children: (obj: {
+    isLoggingIn: boolean;
+    changePassword: (pw: { newPassword: string; confirmNew: string; oldPassword: string }, valid: boolean) => void;
```

Comment:
I think changePassword should only accept a string with new password as an argument. The front-end validation should be performed in the form itself (not allowing to call changePassword at all).

Reply:
I thought `LoginCtrl` could be used in another context. Or should it just be for the LoginPage? It feel a bit weird to just set `oldPassword` to `admin` in the `LoginCtrl`.

Reply:
It still could be used in another context. It's just a matter of what data you pass into the method. Why would you need the `newPassword` and and `confirmNew` when those are the same (I assume this is already assured by ty the time the change password button is clicked)? So my suggestion here is to change the signature of changePassword to `changePassword(password: string)` and make it front end validation independent.